cli/src/chains_config/mod.rs

```diff
@@ -0,0 +1,119 @@
+use std::collections::HashMap;
+use std::fs;
+use std::path::{Path, PathBuf};
+
+use anyhow::{bail, Result};
+use serde::{Deserialize, Serialize};
+
+use crate::config::{Chain, Verifier};
+use crate::opts::ChainsOpts;
+use crate::AppConfig;
+
+#[derive(Clone, Serialize, Deserialize, Debug)]
+pub(crate) struct ConfigTemplate {
+    pub(crate) data_file: Option<PathBuf>,
+    pub(crate) public_dir: PathBuf,
+    pub(crate) qr_dir: PathBuf,
+    pub(crate) verifier: Verifier,
+    pub(crate) chains: HashMap<String, ChainTemplate>,
+}
+
+#[derive(Clone, Serialize, Deserialize, Debug)]
+pub(crate) struct ChainTemplate {
+    pub(crate) name: String,
+    pub(crate) color: String,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub(crate) struct ChainJSON {
+    pub(crate) name: String,
+    pub(crate) nodes: Vec<ChainNode>,
+    pub(crate) icon: String,
+    pub(crate) options: Option<Vec<String>>,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub(crate) struct ChainNode {
+    pub(crate) name: String,
+    pub(crate) url: String,
+}
+
+const EXCLUDE_CHAINS: [&str; 7] = [
+    "Polkadot",
+    "Kusama",
+    "Westend",
+    "Moonbeam",
+    "Moonriver",
+    "Moonbase Relay Testnet",
+    "Arctic Relay Testnet",
+];
+
+pub(crate) fn update_chains_config(chains_opts: ChainsOpts) -> Result<()> {
+    let template_path = Path::new("config-template.toml");
+    let config_template_toml = fs::read_to_string(&template_path)?;
+    let config_template = toml::from_str::<ConfigTemplate>(config_template_toml.as_str())?;
+
+    let chain_params = match chains_opts.env.as_str() {
+        "dev" => (
+            "config_dev.toml",
+            "public/data_dev.json",
+            format!(
+                "https://raw.githubusercontent.com/nova-wallet/nova-utils/master/chains/{}/{}",
+                chains_opts.version, "chains_dev.json"
+            ),
+        ),
+        "prod" => (
+            "config.toml",
+            "public/data.json",
+            format!(
+                "https://raw.githubusercontent.com/nova-wallet/nova-utils/master/chains/{}/{}",
+                chains_opts.version, "chains.json"
+            ),
+        ),
+        _ => bail!("Unknown env. Should be dev or prod"),
+    };
+
+    let chains_json_response = reqwest::blocking::get(chain_params.2).unwrap();
+    let chains_json: Vec<ChainJSON> = chains_json_response.json().unwrap();
+
+    let mut chains: Vec<Chain> = vec![];
+    for chain in chains_json {
+        if EXCLUDE_CHAINS.contains(&chain.name.as_str()) {
+            continue;
+        }
+        let chain_template = config_template.chains.get(&chain.name);
```

Comment:
if I understand the implementation correctly, when a new network is added to nova-utils, it will not be added to metadata-portal until the template is updated?

Reply:
Right. However if there is no network in the template the job will be failing. So we'll be notified that the network has to be added into the template.